### PR TITLE
Refactored into a package.

### PR DESCRIPTION
**Pull request by evetion**
I've put the normal package structure in here (testing, docs, module etc).
Forced to use LF line endings, not the Windows ones, by using .gitattributes and .editorconfig
Made exports so that the moved `scripts/groundwater2d3d.jl` could run, but you could export more.

From within Julia you can do `]activate .` or from the commandline you can do things like `julia --project=. scripts/groundwater2d3d.jl`.

If this works, we can move to have actual GPU unittests over at Gitlab or similar.

### CODE REVIEW

**Comment by visr** [comment]
Looks good to me. Though if you prefer the name Athesis, perhaps it is a good idea to rename the package and repository to that at this point.

Probably worth pointing out that many of the changes here are made automatically, in particular (I assume) https://github.com/invenia/PkgTemplates.jl to set up the package structure including the supporting CI files and such, and https://github.com/domluna/JuliaFormatter.jl was used to auto format the code (this is also integrated into Juno and VSCode).

**Comment by evetion** [comment]
> Probably worth pointing out that many of the changes here are made automatically, in particular (I assume) https://github.com/invenia/PkgTemplates.jl to set up the package structure including the supporting CI files and such, and https://github.com/domluna/JuliaFormatter.jl was used to auto format the code (this is also integrated into Juno and VSCode).

Absolutely, the most work was removing all includes and putting them in the correct order in the Porteau.jl file.

File: src/athesis/kernels_pressure_equation.jl [hunk 1]
```diff
@@ -1,19 +1,18 @@
...
-include("operators.jl")
 
 function averageK(K, i, j, k)
-    Kw = harmonic_mean(K[i-1,j  ,k  ],K[i  ,j  ,k  ])
-    Ke = harmonic_mean(K[i  ,j  ,k  ],K[i+1,j  ,k  ])
-    Ks = harmonic_mean(K[i  ,j-1,k  ],K[i  ,j  ,k  ])
-    Kn = harmonic_mean(K[i  ,j  ,k  ],K[i  ,j+1,k  ])
-    Kb = harmonic_mean(K[i  ,j  ,k-1],K[i  ,j  ,k  ])
-    Kt = harmonic_mean(K[i  ,j  ,k  ],K[i  ,j  ,k+1])
+    Kw = harmonic_mean(K[i - 1,j  ,k  ], K[i  ,j  ,k  ])
```
Comment (mjr-deltares) [inline src/athesis/kernels_pressure_equation.jl:5]: Could we somehow suppress this formatting of indices? It doesn't seem to improve the readability in this case, wouldn't you agree? (edit: this comment should have been part of the review below...)

File: src/athesis/kernels_pressure_equation.jl [hunk 2]
```diff
@@ -26,16 +25,16 @@ end
...
+    # dim = (1,2,3)
 
-    Kw, Ke, Ks, Kn, Kb, Kt = averageK(K,i, j, k)
+    Kw, Ke, Ks, Kn, Kb, Kt = averageK(K, i, j, k)
 
-    F = (1.0/Δx*Δx) * (Ke * (h[i+1,j  ,k  ] - h[i,j,k]) - Kw * (h[i,j,k] - h[i-1,j  ,k  ])) +
-        (1.0/Δy*Δy) * (Kn * (h[i  ,j+1,k  ] - h[i,j,k]) - Ks * (h[i,j,k] - h[i  ,j-1,k  ])) +
-        (1.0/Δz*Δz) * (Kt * (h[i  ,j  ,k+1] - h[i,j,k]) - Kb * (h[i,j,k] - h[i  ,j  ,k-1])) +
-        source[i,j,k]/(Δx*Δy*Δz)
+    F = (1.0 / Δx * Δx) * (Ke * (h[i + 1,j  ,k  ] - h[i,j,k]) - Kw * (h[i,j,k] - h[i - 1,j  ,k  ])) +
```
Comment (mjr-deltares) [inline src/athesis/kernels_pressure_equation.jl:33]: Same argument about the formatting here, I think it less transparent this way

File: src/athesis/operators.jl [hunk 3]
```diff
@@ -6,81 +6,81 @@
...
-@inline div2dᵘ(f,i,j,(Δx,Δy)) = δfδx_2dᵘ(f,i,j,Δx) + δfδy_2dᵘ(f,i,j,Δy)
+@inline div2dᵘ(f, i, j, (Δx, Δy)) = δfδx_2dᵘ(f, i, j, Δx) + δfδy_2dᵘ(f, i, j, Δy)
 # downwind divergence
-@inline div2dᵈ(f,i,j,(Δx,Δy)) = δfδx_2dᵈ(f,i,j,Δx) + δfδy_2dᵈ(f,i,j,Δy)
+@inline div2dᵈ(f, i, j, (Δx, Δy)) = δfδx_2dᵈ(f, i, j, Δx) + δfδy_2dᵈ(f, i, j, Δy)
 
 ############### 3d divergence operators ##################
 # central divergence
-@inline div3dᶜ(f,i,j,k,(Δx,Δy,Δz)) = (δfδx_3dᶜ(f,i,j,k,Δx) + δfδy_3dᶜ(f,i,j,k,Δy) + δfδz_3dᶜ(f,i,j,k,Δz))
+@inline div3dᶜ(f, i, j, k, (Δx, Δy, Δz)) = (δfδx_3dᶜ(f, i, j, k, Δx) + δfδy_3dᶜ(f, i, j, k, Δy) + δfδz_3dᶜ(f, i, j, k, Δz))
```
Comment (mjr-deltares) [inline src/athesis/operators.jl:75]: Maybe it is not possible, but here we might want to have the option to leave out the whitespace for compactness

**Review by mjr-deltares — COMMENTED** [review]
It's nice to have the CI setup now! @evetion Did you look at the failing Travis builds? I saw stuff along the lines of "ERROR: LoadError: InitError: InexactError: check_top_bit(UInt32, 4127983232)". Is this something that you are planning to fix within this PR, is there something that should be changed in the code for this?

**Comment by mjr-deltares** [comment]
Hi @evetion , can we somehow integrate your proposed changes without the style changes (and apply those later to avoid the conflicts)

**Comment by mjr-deltares** [comment]
and I like it that you've setup the CI for this project, want to use it ASAP. Maybe we could discuss what is needed to fix the Travis build.

**Comment by mjr-deltares** [comment]
these changes have mostly been implemented by now, closing this

**Pull request closed without merging**